Assets/MixedRealityToolkit/MSBuild/Scripts/ReferenceItemInfo.cs

```diff
@@ -0,0 +1,65 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+#if UNITY_EDITOR
+using System;
+using System.Collections.Generic;
+using UnityEditor;
+
+namespace Microsoft.MixedReality.Toolkit.MSBuild
+{
+    /// <summary>
+    /// A common base class for reference items such as C# Projects and DLLs to be added to MSBuild.
+    /// </summary>
+    public class ReferenceItemInfo
+    {
+        protected readonly IEnumerable<CompilationPlatformInfo> availablePlatforms;
+
+        /// <summary>
+        /// Gets the Guid associated with the reference.
+        /// </summary>
+        public Guid Guid { get; }
+
+        /// <summary>
+        /// Gets the output path to the reference.
+        /// </summary>
+        public Uri ReferencePath { get; }
+
+        /// <summary>
+        /// Gets name of the reference item.
+        /// </summary>
+        public string Name { get; }
+
+        /// <summary>
+        /// </summary>
+        /// <remarks>
+        /// In the editor, we can support all patforms if it's a pre-defined assembly, or an asmdef with Editor platform checked. 
```

Comment:
>patforms [](start = 46, length = 8)

Nit: platforms

Reply:
Thought also patforms I can get behind

---
In reply to: [301789756](https://github.com/microsoft/MixedRealityToolkit-Unity/pull/5215#discussion_r301789756) [](ancestors = 301789756)